jsapp/js/stores.es6

```diff
@@ -396,8 +396,10 @@ var selectedAssetStore = Reflux.createStore({
 var userExistsStore = Reflux.createStore({
   init () {
     this.checked = {};
-    this.listenTo(actions.misc.checkUsername.completed, this.usernameExists);
-    this.listenTo(actions.misc.checkUsername.failed, this.usernameDoesntExist);
+    if (navigator.onLine) {
```

Comment:
By checking it here - during initialization - it would prevent it from working when the internet connection is regained. I think it would be best for now to add this check `if (navigator.onLine)` in the `notifyUnknownUser` function (in both `jsapp/js/components/permissions/userAssetPermsEditor.es6` and `jsapp/js/components/permissions/userCollectionPermsEditor.es6`) with a comment of explanation. We would need to think of a more general offline mode

Reply:
I tried your solution and it works when the network goes offline, but after regaining connection entering an existing user prompts the `notifyUnkownUser` again, the user would have to refresh the page in order to get expected functionality again. Maybe we should leave this for a more general soltution?

Reply:
I think we need to change `stores.userExists` to not store any information on checks if there is no internet. So both `usernameExists` and `usernameDoesntExist` callbacks should run `this.checked[username] = …` only if there is internet. This will make the user check be triggered and end up being `undefined` (so "unknown").

I did also create a general network failure handling issue https://github.com/kobotoolbox/kpi/issues/2685